### PR TITLE
Javascript - Weather App - PR #1

### PR DESCRIPTION
**Pull request by Rhelli**
> This PR is for Microverse TSE review

# 🌦️🔮 Weather App - my.weather

Hi there TSEs! Here is my pull request for my finished weather forecasting app (from the Javascript module v1.0). As a quick breakdown for functionality available, the app allows user to:

 - Displays current general weather forecast, weather details (i.e. uv index, wind speed, etc), datetime and location info;
 - Automatic background image matching depending on day/night cycle and current weather type;
 - A shortcut bar giving option to refresh current weather, delete all stored locations and switch between Fahrenheit and Celsius;
 - Ability to store multiple locations and save the selected temperature format;
 - Error handling when an unknown location is added.

For a preview of the website, please [head here](https://rhelli.github.io/my.weather/).

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -18,6 +18,10 @@
     "arrow-parens": "off",
     "no-plusplus": "off",
     "consistent-return": "off",
+    "no-unused-vars": "off",
+    "no-undef": "off",
+    "func-names": "off",
+    "no-restricted-globals": "off"
```
Comment (addod19) [inline .eslintrc.json]: Do not make changes to the config files. Kindly check this [link](https://github.com/microverseinc/linters-config/blob/master/javascript/.eslintrc.json) for the required file

File: package.json [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
+  "homepage": "https://Rhelli.github.io/API-Weather-App",
   "version": "1.0.0",
   "description": "A weather app for all your favourite places on earth.",
   "main": "index.js",
   "scripts": {
+    "predeploy": "npm run build",
+    "deploy": "gh-pages -d dist",
     "test": "echo \"Error: no test specified\" && exit 1",
     "build": "webpack",
     "watch": "webpack --watch",
```
Comment (addod19) [inline package.json]: Let us kindly improve our script by using this as inspiration :smile: 
```
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  },

```

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes:recycle: :traffic_light: 

Hi @Rhelli, great job with this project :+1: The UI looks great :100: README file is professional :fireworks: The code structure is awesome and the is divided into specific functions :slightly_smiling_face: However, there are a few issues we need to address so we can move on. Kindly check the comments below:
- I tried to retrieve data from three different locations but no data was returned, also there was no background change as in the screenshot below
vf
This is the console log when making the API request
log
- Do not make changes to the linter files. We currently have some modifications

Once these are fixed, kindly submit another code review :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](linkedin.com/in/daniel-larbi-addo/)

**Comment by Rhelli** [comment]
# Javascript - Weather App - Re-Review # 1

Following the previous TSEs requests, I have modified my project to include the following changes:

 - I have removed all additional ESLint rules and returned eslintrc.json to its original Microverse format.
 - I've fixed the remaining issues that arose from removing those ESLint rules.
 - I've amended my `package.json` files `scripts` to include more development/production specific modes for `dev`, `build` and `start`.
 
Additionally, I would like to point out that the issue occurring where locations were not loading and the API request was not being accepted appears to have been a local error/an error with accessing the API, versus an issue with the code. The API can be a little slow sometimes in returning the data but 99% of the time it is returned without an error (unless an incorrect location is entered). 

I've tested the API calls on 2 machines, both the computer it was built on and another which it was not, and both running locally on both the `dev-server` as well as just running `npm run build` and viewing the project via the full path in Chrome. I do believe the code is executing correctly, however if the problem persists, it would most likely appear to be a problem with the local environment variables (I'm not sure that cloning this project in order to test API calls would therefore be an appropriate way of testing?).

Do reach out to me on Slack if you run into anymore problems though and I'm sure we can get the issue resolved quickly.

Thanks,

💂🏽‍♂️ Rory

**Review by jofer86 — APPROVED** [review]
## `const PROJECT_STATUS = "APPROVED"` 🚀 🚀 🚀 
## :wave: :wave: Hello! I'm Jorge, and  I will be conducting this code review. Let me get a closer look 🔎 🔍 🔎 🔍 🕵🏼 
- Great job!! Project looks amazing and the code is properly structured. I tried your code with my own API keys and all seems to be working fine. You did well not to share your env file, but for review purposes lets just mock the env constant with a JSON file or something like that. Unless you used a credit card for getting an API key, then you better keep them :P, otherwise to avoid complications you should send the API key so that the TSE is able to revew your project correctly. Other than that all good!!
--------------------------------------------------------------------
**Rock on**  🤘🏼 
**_Jorge Fernando Rincón Figueroa._** 
Technical Support Engineer.

**Comment by Rhelli** [comment]
Hi @jofer86, Thanks for the review! I'll make sure to add the API .env (if not security sensitive) in any future projects. It wasn't immediately obvious from the project spec or microverse requirements as to what the procedure for handling the local env was. Nevermind though!

**Pull request merged**